tests/test_configuration.py

```diff
@@ -0,0 +1,108 @@
+import json
+from os import environ
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from lbz.aws_ssm import SSM
+from lbz.configuration import ConfigValue, EnvValue, SSMValue
+from lbz.exceptions import ConfigurationMissingError, ConfigurationParsingError
+
+
+class MyConfigValue(ConfigValue):
+    def getter(self) -> str:
+        return "test-value"
+
+
+class MyNoneConfigValue(ConfigValue):
+    def getter(self) -> None:
+        return None
+
+
+class TestBaseConfig:
+    def test_if_getter_was_used(
+        self,
+    ) -> None:
+        cfg = MyConfigValue("key")
+
+        assert cfg.value == "test-value"
+
+    def test_returns_default_when_no_value_set(self) -> None:
+        cfg = MyNoneConfigValue("test", default=42, parser=int)
+
+        assert cfg.value == 42
+
+    def test_raises_missing_congiration_when_value_is_none(self) -> None:
+        cfg = MyNoneConfigValue("test")
+
+        with pytest.raises(ConfigurationMissingError, match="'test' was not defined."):
+            cfg.value  # pylint: disable=pointless-statement
+
+    def test_if_parser_was_used(self) -> None:
+        class MyIntConfigValue(ConfigValue):
+            def getter(self) -> int:
+                return 1
+
+        cfg = MyIntConfigValue("key")
+
+        assert cfg.value == "1"
+
+    @patch.object(MyConfigValue, "getter", autospec=True)
+    def test_if_getter_was_used_only_once(self, mocked_getter: MagicMock) -> None:
+        mocked_getter.return_value = "42"
+        cfg = MyConfigValue("key")
+
+        assert cfg.value == "42"
+        assert cfg.value == "42"
+
+        mocked_getter.assert_called_once()
+
+    def test_raises_if_parser_failed(self) -> None:
+        class MyFailingConfigValue(ConfigValue):
+            def getter(self) -> tuple:
+                return 1, 2
+
+        cfg = MyFailingConfigValue("key", parser=int)
+
+        with pytest.raises(
+            ConfigurationParsingError,
+            match="'key' could not be parsed with '<class 'int'>'",
```

Comment:
Please, don't create multiline context managers, you can always declare a variable for the expected message.